backend/api/blueprints/leads.py

```diff
@@ -94,6 +94,11 @@ def _get_all_leads(request):
     with db.get_connection() as conn:
         row = conn.execute(select_id_for_tag, tag=tag).first()
         tag_id = row[0] if row else None
+        if tag is not None and tag_id is None:
```

Comment:
This works great. I like that it exits early to avoid making extra database queries.